common/components/WalletDecrypt/Keystore.jsx

```diff
@@ -1,63 +1,115 @@
-  constructor(props) {
-    super(props);
-  }
-
-  handleFileSelection = event => {
-    const fileReader = new FileReader();
-    const inputFile = event.target.files[0];
+export type KeystoreValue = {
+  file: string,
+  password: string,
+  valid: boolean
+};
 
-    fileReader.onload = () => {
-      try {
-        const keyStoreString = fileReader.result;
-        const decryptedWallet = wallet.fromV3(
-          keyStoreString,
-          'asdfasdfasdf',
-          true
-        );
-        const privateHex = ethUtil.bufferToHex(decryptedWallet._privKey);
-        const publicHex = ethUtil.bufferToHex(
-          ethUtil.privateToAddress(decryptedWallet._privKey)
-        );
-        console.log(privateHex, publicHex); // TODO: Remove console log, it's only here to let Travis pass
-      } catch (e) {
-        console.error('Could not parse Keystore file.', e);
-      }
-    };
+export default class KeystoreDecrypt extends Component {
+  props: {
+    value: KeystoreValue,
+    onChange: (value: KeystoreValue) => void,
+    onUnlock: () => void
+  };
 
-    fileReader.readAsText(inputFile, 'utf-8');
+  state = {
+    isPassRequired: false
   };
 
   render() {
+    const { file, password } = this.props.value;
+
     return (
       <section className="col-md-4 col-sm-6">
         <div id="selectedUploadKey">
-          <h4>{translate('ADD_Radio_2_alt')}</h4>
+          <h4>
+            {translate('ADD_Radio_2_alt')}
+          </h4>
 
           <div className="form-group">
             <input
+              className={'hidden'}
               type="file"
               id="fselector"
               onChange={this.handleFileSelection}
             />
-            <label htmlFor="fselector">
+            <label htmlFor="fselector" style={{ width: '100%' }}>
               <a
-                className="btn-file marg-v-sm"
+                className="btn btn-default btn-block"
                 id="aria1"
                 tabIndex="0"
                 role="button"
               >
                 {translate('ADD_Radio_2_short')}
               </a>
             </label>
+            <div
+              className={
+                file.length && this.state.isPassRequired ? '' : 'hidden'
+              }
+            >
+              <p>
+                {translate('ADD_Label_3')}
+              </p>
+              <input
+                className={`form-control ${password.length > 0
+                  ? 'is-valid'
+                  : 'is-invalid'}`}
+                value={password}
+                onChange={this.onPasswordChange}
+                onKeyDown={this.onKeyDown}
+                placeholder={translate('x_Password')}
+                type="password"
+              />
+            </div>
           </div>
         </div>
       </section>
     );
   }
+
+  onKeyDown = (e: SyntheticKeyboardEvent) => {
+    if (e.keyCode === 13) {
+      e.preventDefault();
+      e.stopPropagation();
+      this.props.onUnlock();
+    }
+  };
```

Comment:
Much simpler than this is to listen to and preventDefault a `<form onSubmit>` event, surrounding the input. It's better for accessibility, and makes it easier to have a submit button in the future (One event binding instead of onClick and onKeyDown.)

I notice this around a fair amount in the app though, so feel free to stick with it for consistency now. If I ever feel strongly enough about it, I'd just do a PR to clean it up everywhere.